zebra/zebra_dplane.c

```diff
@@ -2549,6 +2549,8 @@ enum zebra_dplane_result dplane_lsp_add(zebra_lsp_t *lsp)
 	enum zebra_dplane_result ret =
 		lsp_update_internal(lsp, DPLANE_OP_LSP_INSTALL);
 
+	zebra_mpls_label_created(lsp->ile.in_label);
```

Comment:
so at this point, nothing has actually been done - nothing has been programmed. is this the right place to notify interested parties? I'd prefer that anyone who relies on dataplane changes react when the results arrive and the actual status is known - that's how we handle interactions like route redistribution, reachability changes, etc. I think that might also let you remove this hook machinery completely - the zebra_mpls results-handling code could call an sr api directly.